tests/functional/object/mcg/test_bucket_policy.py

```diff
@@ -271,6 +272,12 @@ def test_object_actions(self, mcg_obj, bucket_factory):
         put_policy = put_bucket_policy(mcg_obj, obc_obj.bucket_name, bucket_policy)
         logger.info(f"Put bucket policy response from Admin: {put_policy}")
 
+        # Hardcoded sleep is needed because we lack a confirmation mechanism
+        # we could wait for - even the get-policy result has been observed to be
+        # unreliable in confirming whether they policy is actually taking effect
+        logger.info("Waiting for 60 seconds for the policy to take effect")
+        time.sleep(60)
```

Comment:
Instead of waiting for 60 secs directly, can we break it to 12 retries. So that we don't need to wait 60 secs all the time. WDYT @sagihirshfeld

Reply:
Usually I prefer using retry where we can, but in this case if the policy fails to deny the deletion in time then the second retry attempt would fail regardless because the object would be missing.

We could modify the deletion part of the test so it generates and puts a new object before each attempt, but it's a bit of an over-complication of a simple test case in my opinion.